### DISCUSSION TITLE
Error Select Multiple (Array to String Conversion)

### DISCUSSION THREAD

**Discussion by fahmiSD** [filamentphp/filament#14886] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.52

### How can we help you?

Hello, I'm new to Filament, and I'm trying to use a Select form with the multiple option.

The column in my database is already of type JSON, and here's the form field I'm using:

```
Forms\Components\Select::make('penempatan')
    ->multiple()
    ->options([
        'highlight' => 'Highlight',
        'pop-up' => 'Pop-up',
    ]);
```
In my model, I’ve also cast the penempatan attribute as an array:
```
protected $casts = [
    'penempatan' => 'array',
];
```
Despite this setup, I still get the following error when trying to create a new record:
![image](https://github.com/user-attachments/assets/2cc37a54-bb41-49f4-a007-ba588c71f9df)

However, when I try to edit an existing record, everything works perfectly.

**Comment by leandrocfe** [comment]
Weird. It seems like the cast is being ignored. You could [use hooks](https://filamentphp.com/docs/3.x/panels/resources/creating-records#lifecycle-hooks) to verify if the values (`$this->data`) are correct before saving.

**Reply by fahmiSD** [reply]
thanks, i'll try debugging with the hook.

**Comment by fahmiSD** [comment, marked as answer]
After some experimentation, I found a solution that worked for me. Instead of relying on $casts to handle the penempatan attribute as an array, I manually added an accessor and mutator in my model:
```

public function setPenempatanAttribute($value)
{
    $this->attributes['penempatan'] = json_encode($value);
}

public function getPenempatanAttribute($value)
{
    return json_decode($value, true);
}

```

I also removed the $casts entry for penempatan. This way, the attribute is properly stored as JSON in the database and retrieved as an array in the application.

This approach gives me more control over the serialization process, especially when $casts doesn’t work as expected. Just make sure to validate the data format before saving to avoid encoding/decoding issues.
